### ISSUE TITLE
diag for a permil matrix multiplies in the permils and returns a unitless quantity

### ISSUE THREAD

**Issue by b-r-hamilton** [ggebbie/UnitfulLinearAlgebra.jl#77]
MWE
```
julia> using Unitful, UnitfulLinearAlgebra, LinearAlgebra

julia> mat = UnitfulMatrix(ones(2,2).*u"permille")
2×2 UnitfulMatrix{Float64,2}
 1.0 ‰  1.0 ‰
 1.0 ‰  1.0 ‰

julia> diag(mat)
2-element Vector{Float64}:
 0.001
 0.001

julia> 
```

**Comment by b-r-hamilton** [comment]
works fine for quantities like Kelvin 
```
julia> mat = UnitfulMatrix(ones(2,2).*u"K")
2×2 UnitfulMatrix{Float64,2}
 1.0 K  1.0 K
 1.0 K  1.0 K

julia> diag(mat)
2-element Vector{Quantity}:
 1.0 K
 1.0 K
```

**Comment by ggebbie** [comment]
made 2 fixes, full accounting in comment for PR #80 

please check that it works for you

```julia
julia> using Unitful, UnitfulLinearAlgebra, LinearAlgebra

julia> permil = Unitful.FixedUnits(u"permille")
‰

julia> K = u"K"
K

julia> mat = UnitfulMatrix(ones(2,2)*permil*K)
2×2 UnitfulMatrix{Float64,2}
 1.0 K ‰  1.0 K ‰
 1.0 K ‰  1.0 K ‰

julia> diag(mat)
2-element Vector{Quantity}:
 1.0 K ‰
 1.0 K ‰
```
